### PR TITLE
add keep alive connection support for sending command to raft server

### PR DESCRIPTION
**Pull request by qiuyesuifeng**
#42

### CODE REVIEW

File: server/node_conn.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+	return &nodeConn{
+		conn:        conn,
+		touchedTime: time.Now()}, nil
+}
+
+type createConnFunc func(addr string) (*nodeConn, error)
+
+var defaultConnFunc = func(addr string) (*nodeConn, error) {
+	return newNodeConn(addr)
+}
```
Comment (disksing) [inline server/node_conn.go]: var defaultConnFunc = newNodeConn

File: server/node_conn.go [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+	return conn, nil
+}
+
+// SetIdleTimeout sets idleTimeout of each conn.
+func (ncs *nodeConns) SetIdleTimeout(idleTimeout time.Duration) {
+	ncs.idleTimeout.Set(idleTimeout)
+}
+
+// GetConn gets the conn by addr.
+func (ncs *nodeConns) GetConn(addr string) (*nodeConn, error) {
```
Comment (disksing) [inline server/node_conn.go]: It seems that two concurrent goroutines may take out a same Conn and write to it at the same time?

Comment (siddontang) [inline server/node_conn.go]: no, maybe yes. If two different clusters have a same node.

@qiuyesuifeng  We should put nodeConns in Cluster, not Server.

File: server/node_conn.go (hunk 1 above)
Comment (qiuyesuifeng) [inline server/node_conn.go]: Yes, good suggestion.

File: server/node_conn.go (hunk 2 above)
Comment (qiuyesuifeng) [inline server/node_conn.go]: @disksing  @siddontang  Yes, binds to cluster is a correct choice.

**Comment by qiuyesuifeng** [comment]
PTAL

**Comment by disksing** [comment]
LGTM

**Comment by siddontang** [comment]
LGTM

**Pull request merged**